pulp_container/tests/functional/api/test_flatpak.py

```diff
@@ -0,0 +1,52 @@
+"""Tests that verify Flatpak support"""
+import pytest
+import subprocess
+
+from django.conf import settings
+
+from pulp_container.tests.functional.constants import REGISTRY_V2
+
+
+def test_flatpak_install(
+    add_to_cleanup,
+    registry_client,
+    local_registry,
+    container_namespace_api,
+):
+    if not settings.FLATPAK_INDEX:
+        pytest.skip("This test requires FLATPAK_INDEX to be enabled")
+
+    image_path1 = f"{REGISTRY_V2}/pulp/oci-net.fishsoup.busyboxplatform:latest"
+    registry_client.pull(image_path1)
+    local_registry.tag_and_push(image_path1, "pulptest/oci-net.fishsoup.busyboxplatform:latest")
+    image_path2 = f"{REGISTRY_V2}/pulp/oci-net.fishsoup.hello:latest"
+    registry_client.pull(image_path2)
+    local_registry.tag_and_push(image_path2, "pulptest/oci-net.fishsoup.hello:latest")
+    subprocess.check_call(
+        ["flatpak", "--user", "remote-add", "pulptest", "oci+" + settings.CONTENT_ORIGIN]
+    )
+    # See <https://pagure.io/fedora-lorax-templates/c/cc1155372046baa58f9d2cc27a9e5473bf05a3fb>
+    # "lorax-embed-flatpaks.tmpl: Run the flatpak-install under dbus-run-session" for the need for
+    # dbus-run-session to avoid "error: Cannot autolaunch D-Bus without X11 $DISPLAY":
+    subprocess.check_call(
+        [
+            "dbus-run-session",
+            "flatpak",
+            "-vv",
+            "--ostree-verbose",
+            "--user",
+            "install",
+            "--noninteractive",
+            "pulptest",
+            "net.fishsoup.Hello",
+        ]
+    )
+
+    # Cleanup:
+    subprocess.run(["flatpak", "--user", "uninstall", "--noninteractive", "net.fishsoup.Hello"])
+    subprocess.run(
+        ["flatpak", "--user", "uninstall", "--noninteractive", "net.fishsoup.BusyBoxPlatform"]
+    )
+    subprocess.run(["flatpak", "--user", "remote-delete", "pulptest"])
+    namespace = container_namespace_api.list(name="pulptest").results[0]
+    add_to_cleanup(container_namespace_api, namespace.pulp_href)
```

Comment:
Please, move the cleanup right after `local_registry.tag_and_push(image_path2, "pulptest/oci-net.fishsoup.hello:latest")`. The cleanup is executed when the test finishes: https://github.com/pulp/pulpcore/blob/bbb30b7a5de3cab3ac13e1913facb33f1292e2fe/pulpcore/tests/functional/__init__.py#L926. In case we fail again when running the flatpak application, other tests shall not be affected.

Reply:
done in a0ad4ac0a28e9c61dbf3cec17e3e298c96fe8b3f